test/fin_fold.lean

```diff
@@ -0,0 +1,5 @@
+import Std.Data.Fin.Basic
+import Std.Tactic.GuardExpr
+
+#guard Fin.foldl 4 (fun x i => List.cons i x) [] == [3,2,1,0]
+#guard Fin.foldr 4 (fun i x => List.cons i x) [] == [0,1,2,3]
```

Comment:
Not sure how I feel about having a test file for these, when we really want a lemma! I wouldn't insist on the lemma, but I think I'd prefer removing the test file in the meantime.

Reply:
Deleted. It's easy to recreate if necessary.